### PR TITLE
Feature/time table connection

### PR DESCRIPTION
**Pull request by davin4i13**
connected subjects to time table

### CODE REVIEW

File: DTAdmin/CommonNetworkManager.swift [hunk 1]
```diff
@@ -114,6 +115,34 @@ class CommonNetworkManager {
...
+        request.httpMethod = HttpMehtod.GET.rawValue
+        if let cookies = StoreHelper.getCookie() {
+            request.setValue(cookies[Keys.cookie], forHTTPHeaderField: Keys.cookie)
+        }
+        URLSession.shared.dataTask(with: request) { data, response, error in
+            if let error = error {
+                completion(nil, error)
+            } else {
+                if let response = response as? HTTPURLResponse, let data = data {
+                    if response.statusCode == 200 {
```
Comment (iostaf) [inline DTAdmin/CommonNetworkManager.swift:136]: What will happen if status code is not 200?

File: DTAdmin/Storyboard.swift [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import UIKit
+
+extension UIStoryboard {
+    class func stoyboard(by type: StoryboardType) -> UIStoryboard {
+        return UIStoryboard.init(name: type.rawValue, bundle: nil)
+    }
+}
+
+enum StoryboardType: String {
+    case Subject = "Subjects"
```
Comment (iostaf) [inline DTAdmin/Storyboard.swift]: Enum cases should be in camelCase.

File: DTAdmin/TimeTable/TimeTableListViewController.swift [hunk 3]
```diff
@@ -27,13 +30,35 @@ class TimeTableListViewController: ParentViewController, UITableViewDataSource,
...
-            if let error = error {
-                self.showAllert(title: "Error", message: error.localizedDescription, completionHandler: nil)
-            } else {
-                self.timeTableList = timeTableList ?? [TimeTable]()
+        if let subjectID = self.subjectID {
+            self.startActivity()
+            CommonNetworkManager.shared().timeTableSubject(by: subjectID, completion: { (timeTableList, error) in
+                self.stopActivity()
+                if let error = error {
+                    self.showAllert(title: "Error", message: error.localizedDescription, completionHandler: nil)
```
Comment (iostaf) [inline DTAdmin/TimeTable/TimeTableListViewController.swift]: Please use NSLocalizedString for all user faced messages.

File: DTAdmin/TimeTable/TimeTableListViewController.swift [hunk 4]
```diff
@@ -27,13 +30,35 @@ class TimeTableListViewController: ParentViewController, UITableViewDataSource,
...
+                } else {
+                    self.timeTableList = timeTableList ?? [TimeTable]()
+                }
+            })
+        } else if let groupID = self.groupID {
+            self.startActivity()
+            CommonNetworkManager.shared().timeTable(by: groupID, completion: { (timeTableList, error) in
+                self.stopActivity()
+                if let error = error {
+                    self.showAllert(title: "Error", message: error.localizedDescription, completionHandler: nil)
```
Comment (iostaf) [inline DTAdmin/TimeTable/TimeTableListViewController.swift]: Please use NSLocalizedString for all user faced messages.

File: DTAdmin/TimeTable/TimeTableListViewController.swift [hunk 5]
```diff
@@ -27,13 +30,35 @@ class TimeTableListViewController: ParentViewController, UITableViewDataSource,
...
+                } else {
+                    self.timeTableList = timeTableList ?? [TimeTable]()
+                }
+            })
+        } else {
+            self.startActivity()
+            CommonNetworkManager.shared().timeTable { (timeTableList, error) in
+                self.stopActivity()
+                if let error = error {
+                    self.showAllert(title: "Error", message: error.localizedDescription, completionHandler: nil)
```
Comment (iostaf) [inline DTAdmin/TimeTable/TimeTableListViewController.swift]: Please use NSLocalizedString for all user faced messages.

**Review by iostaf — CHANGES_REQUESTED** [review]

File: DTAdmin/CommonNetworkManager.swift (hunk 1 above)
Comment (davin4i13) [reply DTAdmin/CommonNetworkManager.swift:136]: fixed

File: DTAdmin/Storyboard.swift (hunk 2 above)
Comment (davin4i13) [reply DTAdmin/Storyboard.swift]: fixed

File: DTAdmin/TimeTable/TimeTableListViewController.swift (hunk 3 above)
Comment (davin4i13) [reply DTAdmin/TimeTable/TimeTableListViewController.swift]: fixed

File: DTAdmin/TimeTable/TimeTableListViewController.swift (hunk 4 above)
Comment (davin4i13) [reply DTAdmin/TimeTable/TimeTableListViewController.swift]: fixed

File: DTAdmin/TimeTable/TimeTableListViewController.swift (hunk 5 above)
Comment (davin4i13) [reply DTAdmin/TimeTable/TimeTableListViewController.swift]: fixed

**Review by davin4i13 — COMMENTED** [review]
ready for review

**Review by iostaf — DISMISSED** [review]

**Review by iostaf — APPROVED** [review]

**Pull request merged**